src/backend/distributed/planner/relation_restriction_equivalence.c

```diff
@@ -1267,6 +1269,11 @@ AddRteSubqueryToAttributeEquivalenceClass(AttributeEquivalenceClass
 										  PlannerInfo *root,
 										  Var *varToBeAdded)
 {
+	if (IsVarRelOptOuterJoin(root, varToBeAdded))
```

Comment:
I feel like we should have this check once, which is on `AddToAttributeEquivalenceClass`, similar to the top 2 checks we have in that function.